### PR TITLE
Change compose to pass list of props instead of object

### LINKED ISSUE #47: Composing two of the same kind
I realized this when trying to replace [my own render props composer](https://codesandbox.io/s/04w609rvz0) with the one in this library.

Imagine the following scenario:

```js
<Toggle initial={true}>
  {outer => (
    <Toggle initial={false}>
      {inner => (
        <Checkbox checked={outer.on} onChange={outer.toggle} />
        <Checkbox checked={inner.on} onChange={inner.toggle} />
      )}
    </Toggle>
  )}
</Toggle>
```

Given that `Compose` uses the name of the component to build the props, I'm not sure if this is possible to compose.

```js
<Compose
  states={[
    <Toggle initial={true} />,
    <Toggle initial={false} />,
  ]}
>
  // The line below shows the problem, we have a name clash in the prop names
  {({ toggle, toggle }) => (
    <Checkbox checked={toggle.on} onChange={toggle.toggle} />
    <Checkbox checked={toggle.on} onChange={toggle.toggle} />
  )}
</Compose>
```

If `Composed` passed the different props of the composed components as positional arguments to the unified render prop function, this would be solved:

```js
<Compose
  states={[
    <Toggle initial={true} />,
    <Toggle initial={false} />,
  ]}
>
  {(outer, inner) => (
    <Checkbox checked={outer.on} onChange={outer.toggle} />
    <Checkbox checked={inner.on} onChange={inner.toggle} />
  )}
</Compose>
```

If you think about it, it makes more sense, if you provide a sequence of components to be composed, that you get the corresponding sequence of props that each of them provide, instead of an object of props.

If you agree I'm more than happy to make a PR myself.

### PR DESCRIPTION
**Pull request by gnapse**
Fixes #47

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 <br />
 
-> This branch is **unstable** and is in **active development**.  
+> This branch is **unstable** and is in **active development**.
```
Comment (TrySound) [inline README.md]: Did you remove double space in the end?

Comment (gnapse) [reply README.md]: Oh sorry, it probably is my editor configured to remove trailing whitespace. I can add it back if you want.

Comment (renatorib) [reply README.md]: Trailing whitespace in markdown is line break, it's make difference. 😢

Comment (gnapse) [reply README.md]: Ooops, did not know that. I'll fix it and fix my editor settings. Sorry.

**Comment by TrySound** [comment]
@gnapse There's a lot more. Look at diff.

**Comment by gnapse** [comment]
Think I fixed them all now. Sorry. First time I see this markdown feature.

**Comment by renatorib** [comment]
I'll review & merge this soon

**Comment by gnapse** [comment]
Always happy to contribute to open source.

**Pull request merged**